src/IECoreImage/DisplayDriverServer.cpp

```diff
@@ -102,6 +109,34 @@ static std::map<std::string, const DisplayDriverServer::PortRange> g_portRegistr
 
 } // namespace
 
+#if BOOST_VERSION >= 106600
+class DisplayDriverServer::Session : public RefCounted
+{
+	public:
+
+		Session( boost::asio::io_context& io_service, MergeMap& mergeMap );
+        ~Session() override;
+
+        boost::asio::ip::tcp::socket& socket();
+        void start();
+
+    private:
+
+  		void handleReadHeader( const boost::system::error_code& error );
+      	void handleReadOpenParameters( const boost::system::error_code& error );
+        void handleReadDataParameters( const boost::system::error_code& error );
+        void sendResult( DisplayDriverServerHeader::MessageType msg, size_t dataSize );
+        void sendException( const char *message );
+
+    private:
+		boost::asio::ip::tcp::socket m_socket;
+		DisplayDriverPtr m_displayDriver;
+		DisplayDriverServerHeader m_header;
+		CharVectorDataPtr m_buffer;
+  		MergeMap& m_mergeMap;
+		std::optional<int> m_mergeId;
+};
+#else
```

Comment:
It will be too hard to maintain this amount of almost-identical code in separate `#if` blocks. As above, I think we can simply change to `io_context` unconditionally.